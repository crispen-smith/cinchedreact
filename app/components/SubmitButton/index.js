/**
 *
 * SubmitButton
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import colors from '../../resources/colors';

const SB = styled.input`
  font-size: 2rem;
  margin: 1rem auto;
  border-radius: 25%;
  background-color: ${colors.pink};
`;

function SubmitButton(props) {
  return <SB type="submit" value={props.label} />;
}

SubmitButton.propTypes = {
  label: PropTypes.string.isRequired,
};

export default SubmitButton;
