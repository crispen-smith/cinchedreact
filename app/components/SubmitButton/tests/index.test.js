import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import SubmitButton from '../index';

describe('<SubmitButton />', () => {
  it('Renders a disabled SubmitButton when the props include enabled-false', () => {
    const element = (
      <SubmitButton
        label="test"
        enabled={false}
        onClick={() => true}
        dataCursor="pointer"
      />
    );
    const button = renderer.create(element).toJSON();
    expect(button).toMatchSnapshot();

    const sb = mount(element);
    const input = sb.find('input').at(0);

    expect(input.props().disabled).toEqual(true);
  });

  it('Renders a disabled SubmitButton when the props include enabled-true', () => {
    const element = (
      <SubmitButton
        label="test"
        enabled
        onClick={() => true}
        dataCursor="pointer"
      />
    );
    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();

    const sb = mount(element);
    const input = sb.find('input').at(0);

    expect(input.props().disabled).toEqual(false);
  });
});
