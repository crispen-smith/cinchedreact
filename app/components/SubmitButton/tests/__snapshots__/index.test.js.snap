// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmitButton /> Renders a BrandBox 1`] = `
<input
  className="SubmitButton__SB-cMjXsV bpCIXX"
  type="submit"
  value="test"
/>
`;
